tools/cfninclude-rewriter/bin/cfninclude-rewriter.ts

```diff
@@ -0,0 +1,25 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+if (process.argv.length != 4) {
+  process.stderr.write('Usage: cfninclude-rewriter <cfn-include directory> <monolithic package name>');
+  process.exit(1);
+}
+
```

Comment:
Looking at this again, it seems that you are rewriting a previously generated `cfn-types-2-classes.json` file.

Would it be cleaner and robust to re-generate it using part of this script again? - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/cloudformation-include/build.js

What I originally meant with the new build tool was to refactor the production of `cfn-classes-2-type.json` out of the `cloudformation-include` package and run it thrice. Once in cloudformation-include, once in monocdk and once in aws-cdk-lib.

Reply:
I don't think that's a good idea, because the `build.js`script is much more complicated - it traverses the JSII manifests of all CDK modules to search for L1 classes inside of them. That takes a long time , and it would be very wasteful to perform that traversal again for `monocdk` and `aws-cdk-lib` (it would add minutes to this packages already very slow build times). It's also superfluous, as we know that the traversal cannot discover any additional classes over what's in that file already.

Also, the name `cfninclude-rewriter` doesn't make too much sense if you're generating the file, and not rewriting it.